dist/bin/scala

```diff
@@ -27,147 +26,211 @@ if [ -z "$PROG_HOME" ] ; then
   cd "$saveddir"
 fi
 
-addJvmOptions () {
-  jvm_options+=("'$1'")
-}
+source "$PROG_HOME/bin/common"
 
-addScalacOptions () {
-  java_options+=("'$1'")
+# This script operates in one of 3 usage modes:
+#   script
+#   repl
+#   run
+# execute_mode replaces mutually exclusive booleans:
+#  execute_repl=false
+#  execute_run=false
+#  execute_script=false
+setExecuteMode () {
+  case "${execute_mode-}" in
+  "") execute_mode="$1" ; shift ;;
+  *) echo "execute_mode==[${execute_mode-}], attempted overwrite by [$1]" 1>&2
+    exit 1
+    ;;
+  esac
 }
 
-source "$PROG_HOME/bin/common"
 
-declare -a residual_args
-declare -a script_args
-execute_repl=false
-execute_run=false
-execute_script=false
-with_compiler=false
-class_path_count=0
-CLASS_PATH=""
-save_compiled=false
+with_compiler=false             # to add compiler jars to repl classpath
+let class_path_count=0 || true  # count classpath args, warning if more than 1
+save_compiled=false             # to save compiled script jar in script directory
+CLASS_PATH="" || true           # scala classpath
```

Comment:
Same question as above regarding `|| true`?

Reply:
So we can temporarily do manual testing with these two settings:
```bash
set -o nounset
set -o errexit
```
To prevent unsetting of variables from being an error, they become `CLASS_PATH="" || true`.
Changing `"$PROG_HOME"` to `"${PROG_HOME-}" is for the same reason.